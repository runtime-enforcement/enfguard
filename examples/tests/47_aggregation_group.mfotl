ALWAYS (FORALL s, y. s <- MAX(x; y; ONCE (C(x, y))) IMPLIES C(s, y))