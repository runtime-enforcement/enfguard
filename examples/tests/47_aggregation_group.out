The formula
 □[0s,∞) (∀s:int. ∀y:int. (s:int <- MAX(x:int; y:int; ⧫[0s,∞) C(x:int, y:int))) → C(s:int, y:int))
is enforceable and types to
 □[0s,∞) ((∀s:int. (∀y:int. ((s:int <- MAX(x:int; y:int; ⧫[0s,∞) C(x:int, y:int))) →:R (C(s:int, y:int) : Cau) : Cau) : Cau)) : Cau) : Cau
[Enforcer] @1 OK.
[Enforcer] @1 nothing to do proactively.
[Enforcer] @2 OK.
[Enforcer] @2 nothing to do proactively.
[Enforcer] @3 OK.
